### PR TITLE
Music: instructions TTS

### PR DESCRIPTION
**Pull request by breville**
This adds very simple text-to-speech (TTS) support to the Lab2 instructions component, and allows Music Lab levels to use it when `?show-tts=true` is added to the URL.  It uses the Web Speech API that's built in to modern browsers.

Each piece of text and feedback gets a play arrow in the upper-right corner.  When pressed, it cancels any existing TTS and starts the new one.  Markdown is first converted to plain text.

There is more that can be done in follow-up, such as waiting until a voice is ready before initially showing the play button, showing the playback status, maybe allowing the voice to be stopped if it's playing, and choosing the ideal voice for the current environment and language, as begun in AI for Oceans in https://github.com/code-dot-org/ml-activities/pull/403.

### Screenshot

Screenshot 2024-09-10 at 6 50 24 PM

### CODE REVIEW

**Comment by dju90** [comment]
What happens now if you click the play button before a voice is ready? Is there just a delay or does it crash?

File: apps/src/lab2/views/components/TextToSpeech.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const TextToSpeech: React.FunctionComponent<TextToSpeechProps> = ({text}) => {
+  const playText = useCallback(() => {
+    const plainText = markdownToTxt(text);
+    const utterance = new SpeechSynthesisUtterance(plainText);
+    speechSynthesis.cancel();
+    speechSynthesis.speak(utterance);
+  }, [text]);
+
+  return (
+    <div
```
Comment (bencodeorg) [inline apps/src/lab2/views/components/TextToSpeech.tsx]: Can we make this a button?

**Comment by breville** [comment]
> What happens now if you click the play button before a voice is ready? Is there just a delay or does it crash?

Added a check so that we don't attempt to play unless we find a voice that is available for the current language.  We also now emits a message to the console about how many appropriate voices were found.

File: apps/src/lab2/views/components/TextToSpeech.tsx (hunk 1 above)
Comment (breville) [reply apps/src/lab2/views/components/TextToSpeech.tsx]: Does it give additional functionality?  I tried that initially, and while it's good that the shadow on hover no longer needs to be overridden, we'd still need to override the background color, the border, and the border when active.

Comment (bencodeorg) [reply apps/src/lab2/views/components/TextToSpeech.tsx]: Just pulled locally and it appears I'm not actually able to focus (at least visibly) and interact with this via keyboard on Chrome/Safari, do you repro? Works if I switch to `button` element.

Comment (sanchitmalhotra126) [reply apps/src/lab2/views/components/TextToSpeech.tsx]: You should be able to use the [@remove-button-styles](https://github.com/code-dot-org/code-dot-org/blob/6f24c08bf35ef6dc4e6068d5ef353737f3c2ee8b/apps/src/mixins.scss#L3) mixin to remove all the default application button styling.

Also depending on the final styling, we could also use an icon-only DSCO button maybe?

Comment (breville) [reply apps/src/lab2/views/components/TextToSpeech.tsx]: Updated to use a `button`.

**Review by bencodeorg — APPROVED** [review]
nice!

**Comment by breville** [comment]
Some more updates:

- tidied up the rendering of the instructions to have text scrolling independent of the container which holds the play button.
- removed the old, unused "big image" support in the instructions while in there.
- started using the default voice, rather than trying to find the first with the current locale, which seems to work better.
- other languages seem to work better when the language is set on the utterance.

File: apps/src/lab2/views/components/instructions.module.scss [hunk 2]
```diff
@@ -124,25 +122,30 @@
...
 }
 
 .message-light {
   @extend %message;
   background-color: $light_gray_50;
   color: $neutral_dark;
-  border: dashed $light_gray_100 2px;
 }
 
 .buttonInstruction {
```
Comment (bencodeorg) [inline apps/src/lab2/views/components/instructions.module.scss:151]: Kind of a tangent, but I deleted a bunch of this styling, not sure why it's showing up here?

**Review by bencodeorg — APPROVED** [review]
lgtm!

File: apps/src/lab2/views/components/instructions.module.scss (hunk 2 above)
Comment (breville) [reply apps/src/lab2/views/components/instructions.module.scss:151]: As discussed, probably just because latest staging isn't merged into this branch.

**Pull request merged**